Sources/NIOExtras/LengthFieldBasedFrameDecoder.swift

```diff
@@ -0,0 +1,122 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+///
+/// A decoder that splits the received `ByteBuffer` by the number of bytes speicifed in a fixed length header
+/// contained within the buffer.
+/// For example, if you received the following four fragmented packets:
+///     +---+----+------+----+
+///     | A | BC | DEFG | HI |
+///     +---+----+------+----+
+///
+/// Given that the specified header length is 1 byte,
+/// where the first header specifies 3 bytes while the second header speicifies 4 bytes,
+/// a `LengthFieldBasedFrameDecoder` will decode them into the following packets:
+///
+///     +-----+------+
+///     | BCD | FGHI |
+///     +-----+------+
+///
+/// 'A' and 'E' will be the headers and will not be passed forward.
+///
+
+public final class LengthFieldBasedFrameDecoder: ByteToMessageDecoder {
```

Comment:
I have used this class in Netty, where it has this name. I didn't really consider it too much because of that. I was planning to send future pull requests to make a class where there is a fixed length header and the data length specification can be anywhere in that header. That said, I think I prefer the use of 'prefix' and it would still fit. I am happy for you to make a call, it probably matters more to the project that the individual class. Also, maybe "LengthHeaderFrameDecoder".

Reply:
Were you planning to make a *new* class that did this, or extend this one? If you're extending this one, I'm happy for this one to have the name `LengthFieldBasedFrameDecoder`. Otherwise, I think we should change the name of this class, as `LengthFieldBasedFrameDecoder` in `Netty` is the class that lets you select where the length field is in the header.